ui/perfherder/alerts/Assignee.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Input, InputGroup } from 'reactstrap';
+
+export default class Assignee extends React.Component {
+  constructor(props) {
+    super(props);
+    const { assigneeUsername } = props;
+
+    this.state = {
+      assigneeUsername,
+      inEditMode: false,
+      newAssigneeUsername: null,
+    };
+  }
+
+  componentDidMount() {
+    // React's onKeyPress isn't able to listen to 'Escape'.
+    // This is a workaround.
+    document.addEventListener('keydown', this.keydownListener);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.keydownListener);
+  }
+
+  goToEditMode = () => {
+    const { user } = this.props;
+    const { assigneeUsername } = this.state;
+
+    if (user.isStaff) {
+      this.setState({
+        inEditMode: true,
+        // input prefills with this field, so
+        // we must have it prepared
+        newAssigneeUsername: assigneeUsername,
+      });
+    }
+  };
+
+  editUsername = newAssigneeUsername => {
+    this.setState({ newAssigneeUsername });
+  };
+
+  pressedEnter = async event => {
+    if (event.key === 'Enter') {
+      const { onAssigneeChange, updateViewState } = this.props;
+      const newAssigneeUsername = event.target.value;
+
+      const response = await onAssigneeChange(newAssigneeUsername);
+
+      if (response.updatedAlertSummary) {
+        this.setState({
+          assigneeUsername: newAssigneeUsername,
+          inEditMode: false,
+        });
+      } else {
+        return updateViewState({
+          errorMessages: [
+            `Failed to set new assignee "${newAssigneeUsername}"`,
+            ...response.errorMessages,
```

Comment:
Shouldn't `response.errorMessages` be part of the error message?  I think with how you're setting state here using `processResponse` in the `updateAssignee` function doesn't quite work. The intention for that helper was to eliminate the boilerplate of `if (!failureStatus)..`(etc) and just set state directly. Since you're doing it a bit differently I'd just return the response and use your logic here but in corporate the error message into the template literal on line 60.

Reply:
Got it. I've addressed this as suggested.